platform/packages/versioning/src/lib.rs

```diff
@@ -34,47 +77,10 @@ impl Version {
     }
 }
 
-pub fn parse_semver(version: &str) -> SemVer {
```

Comment:
why did you abandon this implementation? It was fine ... very fine ... much more readable and less error-prone than the new code!

Reply:
It isn't const-evaluatable; The reasoning behind the PR is that this should be done during compilation instead of parsing the version on-demand every time.

Reply:
There are plenty of tests that make sure it does in fact parse it correctly.

Reply:
Added more tests, specifically for checking every branch of the parsing.